packages/core/src/memory/types.ts

```diff
@@ -76,7 +78,11 @@ export type MemoryConfig = {
       };
   workingMemory?: WorkingMemory;
   threads?: {
-    generateTitle?: boolean;
+    generateTitle?:
+      | boolean
+      | {
+          model: LanguageModelV1 | ((ctx: RuntimeContext) => LanguageModelV1 | Promise<LanguageModelV1>);
+        };
```

Comment:
@DanielSLew You can simplify this definition as follows. In fact, the `AgentConfig`'s `model` field already uses `DynamicArgument<MastraLanguageModel>`:

```suggestion
    generateTitle?:
      | boolean
      | {
          model: DynamicArgument<MastraLanguageModel>;
        };
```

Refs: [https://github.com/mastra-ai/mastra/blob/main/packages/core/src/agent/types.ts#L48](https://github.com/mastra-ai/mastra/blob/main/packages/core/src/agent/types.ts#L48)